pkg/apis/serving/metadata_validation.go

```diff
@@ -76,3 +78,42 @@ func ValidateTimeoutSeconds(ctx context.Context, timeoutSeconds int64) *apis.Fie
 	}
 	return nil
 }
+
+// ValidateRevisionName validates name and generateName for the revisionTemplate
+func ValidateRevisionName(ctx context.Context, name, generateName string) *apis.FieldError {
+	if generateName != "" {
+		msgs := validation.NameIsDNS1035Label(generateName, true)
```

Comment:
`if msg :=. ...; msg != "" {`

Reply:
1. type of `msgs` is `[]string` so we cannot do `msgs != ""`
2. Added if condition as per suggestion